video/out/vulkan/context_wayland.c

```diff
@@ -36,7 +36,7 @@ static void wayland_vk_swap_buffers(struct ra_ctx *ctx)
 {
     struct vo_wayland_state *wl = ctx->vo->wl;
 
-    if (!wl->opts->wl_disable_vsync)
+    if (!wl->has_commit_timing || !wl->has_fifo || !wl->opts->wl_disable_vsync)
```

Comment:
It's probably unlikely a compositor implements fifo but not commit-timing, but the condition should still be `&&` here

Reply:
Mesa is also not going to use wp-fifo on VRR displays if the compositor does not support wp-presentation v2. In fact, which method mesa uses can change from frame to frame if the user moves the window between displays.

Reply:
If either one of those is missing we definitely have to use the internal heuristic, hence `||`.